projects/chaos/src/state_handler.c

```diff
@@ -0,0 +1,43 @@
+#include "state_handler.h"
+
+#include <stddef.h>
+
+#include "can.h"
+#include "can_msg_defs.h"
+#include "can_unpack.h"
+#include "chaos_events.h"
+#include "event_queue.h"
+
+#define STATE_HANDLER_EMPTY_DATA 0
+
+// CANRxHandlerCb
+static StatusCode prv_handle_power_state_msg(const CANMessage *msg, void *context,
+                                             CANAckStatus *ack_reply) {
+  (void)context;
+  uint8_t power_state = 0;
+  CAN_UNPACK_POWER_STATE(msg, &power_state);
+  // Handler the power state.
+  // TODO(ELEC-105): Replace dummy values with those exported by driver controls.
+  switch (power_state) {
+    case 0:
+      event_raise(CHAOS_EVENT_SEQUENCE_EMERGENCY, STATE_HANDLER_EMPTY_DATA);
```

Comment:
Driver controls shouldn't actually be initiating a transition to the emergency state. That should come from Plutus or Chaos.

Reply:
Oops this should be in a separate PR, this was a very early version of this file. I'll remove it.